spyder/plugins/editor/plugin.py

```diff
@@ -866,85 +866,114 @@ def get_plugin_actions(self):
-                                     [MENU_SEPARATOR] +
-                                     debug_toolbar_actions +
-                                     [MENU_SEPARATOR] +
-                                     edit_toolbar_actions)
-        self.pythonfile_dependent_actions = [run_action, configure_action,
-                                             set_clear_breakpoint_action,
-                                             set_cond_breakpoint_action,
-                                             debug_action, run_selected_action,
-                                             run_cell_action,
-                                             run_cell_advance_action,
-                                             re_run_last_cell_action,
-                                             blockcomment_action,
-                                             unblockcomment_action,
-                                             self.winpdb_action]
+        self.dock_toolbar_actions = (
+            file_toolbar_actions +
+            [MENU_SEPARATOR] +
+            source_toolbar_actions +
+            [MENU_SEPARATOR] +
+            run_toolbar_actions +
+            [MENU_SEPARATOR] +
+            debug_toolbar_actions +
+            [MENU_SEPARATOR] +
+            edit_toolbar_actions
+        )
+        self.pythonfile_dependent_actions = [
+            run_action,
+            configure_action,
+            set_clear_breakpoint_action,
+            set_cond_breakpoint_action,
+            debug_action,
+            run_selected_action,
+            run_cell_action,
+            run_cell_advance_action,
+            re_run_last_cell_action,
+            blockcomment_action,
+            unblockcomment_action,
+            self.winpdb_action
+        ]
         self.cythonfile_compatible_actions = [run_action, configure_action]
-        self.file_dependent_actions = self.pythonfile_dependent_actions + \
-                [self.save_action, save_as_action, save_copy_as_action,
-                 print_preview_action, self.print_action,
-                 self.save_all_action, gotoline_action, workdir_action,
-                 self.close_action, self.close_all_action,
-                 self.toggle_comment_action, self.revert_action,
-                 self.indent_action, self.unindent_action]
+        self.file_dependent_actions = (
+            self.pythonfile_dependent_actions +
+            [
+                self.save_action,
+                save_as_action,
+                save_copy_as_action,
+                print_preview_action,
+                self.print_action,
+                self.save_all_action,
+                gotoline_action,
+                workdir_action,
+                self.close_action,
+                self.close_all_action,
+                self.toggle_comment_action,
+                self.revert_action,
+                self.indent_action,
+                self.unindent_action

```

Comment:
I suggest we leave the trailing comma on the last element as well.

This allows for cleaner DIFFs when having to add new elements. Something like this:
```python
...
                self.indent_action,
                self.unindent_action'
            ]
```

Reply:
This applies to all the other structures above @steff456

Reply:
I don't agree with this. We don't follow that convention and it seems odd to me too (I've never seen it before).

Reply:
I strongly agree; in fact, I think we should adopt the consistent convention for new/sufficiently modified code to write non-inline collections of three or more elements that each fit on one line in the general form [recommended by PEP 8](https://www.python.org/dev/peps/pep-0008/#id17)

```python
my_list = [
    item_one,
    item_two,
    item_three,
    ]
```

(i.e. opening/closing braces not on the same line as the first/last item, and commas after every tiem) given we're currently all over the place in that regard, and this form produces the most readable diffs and (IMO) code as well. I personally prefer ``]`` aligned with the first non-whitespace line of the last list item, as it is more consistent with the other indentation rules and keeps the scope within the assignment as it should be, but PEP 8 also allows the more "traditional" style so I'm not going to raise a fuss about it (though it would be nice to pick one and be consistent).

> self.unindent_action'

I assume you mean ``,``, not ``'``, @goanpeca :)

Reply:
I'm strongly -1 on this, really. It's a purely stylistic stuff that we shouldn't enforce. If PRs authors like to use it, I'm fine with it, but it's up to each one.

So, if there's no more things to address, I'd like to merge this one. @goanpeca?

Reply:
> We don't follow that convention

We don't, but we should. Its [recommended by PEP 8](https://www.python.org/dev/peps/pep-0008/#id29) and the [Google styleguide](https://google.github.io/styleguide/pyguide.html#341-trailing-commas-in-sequences-of-items) among others, and it both plays better with VCSes and avoids silly errors when re-ordering, adding or removing elements from lists.

> (I've never seen it before).

You've never read PEP 8? :) Its used throughout all of the examples there.

Reply:
> It's a purely stylistic stuff that we shouldn't enforce.

So is the rest of PEP 8. This in particular has *objective* benefits with both cleaner diffs and avoiding errors.

Reply:
> You've never read PEP 8? :)

I meant I've never seen it used in the lots of projects I read code from.

Reply:
> So is the rest of PEP 8

We don't follow Pep 8 to the letter, and this is not even suggested by Pep8speaks.

Reply:
> this is not even suggested by Pep8speaks.

Because PEP8Speaks only enforces the strict requirements of PEP 8, and not the recommended things (like 4-space indents, which we *do* enforce).

Reply:
Many projects do use it and many styling tools enforce it as part of their rules. There are also blogpost on why this makes cleaner diffs and easier to read code/cleaner code. Also @CAM-Gerlach has pointed out official resources, that recommend that as well.

Maybe it is a discussion for another thread or separate issue.

---

I have advocated in the past (and I still do) to use tools like https://github.com/google/yapf so we do not have to worry too much about these discussions, commit to a style almost all feel comfortable with (as it is impossible to get a perfect style that fits all of the needs of every developer). It is much easier to run `yapf <file-name>.py` before every commit (or as part of CI) and use the time in coding. Also we can enforce this for every other project we maintain.

>The ultimate goal is that the code YAPF produces is as good as the code that a programmer would write if they were following the style guide. It takes away some of the drudgery of maintaining your code.

Reply:
> We don't follow Pep 8 to the letter

Other than line length in ``icon_manager.py`` which is explicitly justified, I'm not sure where we don't follow a non-optional PEP 8 convention in new code, and our [Style Guide](https://github.com/spyder-ide/spyder/wiki/Dev%3A-Coding-Style), which you approved, says ``Unless otherwise specified, follow PEP 8 specifications and recommendations for coding style`` (and the only PEP 8 exception noted is retaining PyQt method names, which we have to do to override them).

Furthermore, this is orthogonal to my point, which was that we *do* enforce *lots* of other ``purely stylistic stuff``, i.e. anything we do enforce from PEP 8 (it is a *style* guide after all).

Reply:
> Because PEP8Speaks only enforces the strict requirements of PEP 8, and not the recommended things

Exactly, this is just a recommendation, not another thing we should start enforcing.

Reply:
Sure we can merge @ccordoba12.

---

(Still auto formatting code is a discussion we should be having and hopefully implementing. See https://github.com/spyder-ide/spyder/issues/9272)

Reply:
We already do enforce other "recommendations"/"suggestions"/"optional" elements in PEP 8, including using ``()`` for continuations, 4-space indents, 8 spaces for hanging indents of ``if`` statements, etc, and again our style guide explicitly says ``follow PEP 8 specifications and recommendations for coding style``, and this one in particular has objective advantages outside of just style.